Add major/minor axis length and ratio

jsta opened the pull request:
See #4

jhollist commented on R/lakeMinorAxisLength.R:
```diff
@@ -0,0 +1,61 @@
+  myLine <- SpatialLines(list(Lines(list(Line(myLine.min)), "1")),
+              proj4string = CRS(proj4string(inLakeMorpho$lake)))
+
+  result <- max(dist2center)
```
should this be min(dist2center)??

jhollist commented on R/lakeMinorAxisLength.R:
```diff
@@ -0,0 +1,61 @@
+  # https://stackoverflow.com/questions/18278382/how-to-obtain-the-lengths-of-semi-axes-of-an-ellipse-in-r
+  elpshull <- predict(cluster::ellipsoidhull(lakeShorePoints))
+  elpshull.center <- matrix(colMeans((elpshull)), ncol = 2, nrow = 1)
+  
```
Back to the timing on this.  If it is quick we can leave as is, but if it takes a while for larger polys then perhaps we could store both the major/minor axis and the ellipsoid on the lakeMorpho object, thus if it has been run for either major or minor already, we don't have to calculate it again.

jhollist commented on R/lakeMajorAxisLength.R:
```diff
@@ -0,0 +1,60 @@
+  myLine <- SpatialLines(list(Lines(list(Line(myLine.max)), "1")),
+              proj4string = CRS(proj4string(inLakeMorpho$lake)))
+
+  result <- max(dist2center)
```
Is this different that result we'd get from gLength(myLine)?  Not up enough on my ellipsoids/convex hulls to know.  All the other results use gLength to pull this value.

jhollist commented on R/lakeMinorMajorAxisRatio.R:
```diff
@@ -0,0 +1,48 @@
+#'         major and minor axes of the lake. Units are the same as the input data.
+#' 
+#' @references \href{https://en.wikipedia.org/wiki/Semi-major_and_semi-minor_axes}{Link}
+#' @import sp rgeos methods
```
I think we'd want to either use the pkg::function, @import or @importFrom.    Also not sure we need to have all these imports for this function.  Looks like just rgeos is needed.  My current preference is to specify with pkg::function, but I think I used @import in the rest of `lakemorpho`.  One or the other is fine with me.

jsta commented:
Ok, I think I addressed all of your comments with the exception of the limnology references. I know minor-major axis ratio is used in image classification analysis but I am not aware of specific limnology references.

jhollist commented:
@jsta 

Looks great.  Merging shortly.  I *may* get around to pushing the update to CRAN in the next day or so.

Thanks again for making the contribution!

jhollist commented:
FYI @jsta, just submitted to CRAN!
